Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34429&attempt_id=312e2ab9-2030-4702-92d8-9b07bd4005ef)

<details><summary>Partial log (click to expand)</summary><p>

```
[32/94] Compiling C object 'xkbcommon-internal@sta/src_utf8.c.o'.
[33/94] Linking static target libxkbcommon-internal.a.
[34/94] Linking target libxkbcommon.0.dylib.
FAILED: libxkbcommon.0.dylib
clang  -o libxkbcommon.0.dylib  -Wl,-headerpad_max_install_names -shared -install_name @rpath/libxkbcommon.0.dylib -Wl,-force_load libxkbcommon-internal.a -Wl,--version-script=/private/tmp/nix-build-libxkbcommon-0.8.0.drv-0/libxkbcommon-0.8.0/xkbcommon.map
ld: unknown option: --version-script=/private/tmp/nix-build-libxkbcommon-0.8.0.drv-0/libxkbcommon-0.8.0/xkbcommon.map
clang-5.0: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
builder for '/nix/store/n4jwn13fksg6z945waxslmmxsihf49yz-libxkbcommon-0.8.0.drv' failed with exit code 1
error: build of '/nix/store/n4jwn13fksg6z945waxslmmxsihf49yz-libxkbcommon-0.8.0.drv' failed
```
</p></details>

--- a/pkgs/development/libraries/libxkbcommon/default.nix
+++ b/pkgs/development/libraries/libxkbcommon/default.nix
@@ -1,30 +1,35 @@
-{ stdenv, fetchurl, pkgconfig, yacc, flex, xkeyboard_config, libxcb, libX11 }:
+{ stdenv, fetchurl, fetchpatch, meson, ninja, pkgconfig, yacc, xkeyboard_config, libxcb, libX11, doxygen }:
 
 stdenv.mkDerivation rec {
-  name = "libxkbcommon-0.7.2";
+  name = "libxkbcommon-0.8.0";
 
   src = fetchurl {
-    url = "http://xkbcommon.org/download/${name}.tar.xz";
-    sha256 = "1n5rv5n210kjnkyrvbh04gfwaa7zrmzy1393p8nyqfw66lkxr918";
+    url = "https://xkbcommon.org/download/${name}.tar.xz";
+    sha256 = "0vgy84vfbig5bqznr137h5arjidnfwrxrdli0pxyn2jfn1fjcag8";
   };
 
-  outputs = [ "out" "dev" ];
+  outputs = [ "out" "dev" "doc" ];
 
-  nativeBuildInputs = [ pkgconfig ];
-  buildInputs = [ yacc flex xkeyboard_config libxcb ];
+  nativeBuildInputs = [ meson ninja pkgconfig yacc doxygen ];
+  buildInputs = [ xkeyboard_config libxcb ];
 
-  configureFlags = [
-    "--with-xkb-config-root=${xkeyboard_config}/etc/X11/xkb"
-    "--with-x-locale-root=${libX11.out}/share/X11/locale"
+  patches = [
+    # darwin compatibility
+    (fetchpatch {
+      url = https://github.com/xkbcommon/libxkbcommon/commit/edb1c662394578a54b7bbed231d918925e5d8150.patch;
+      sha256 = "0ydjlir32r3xfsbqhnsx1bz6ags2m908yhf9i09i1s7sgcimbcx5";
+    })
   ];
 
-  preBuild = stdenv.lib.optionalString stdenv.isDarwin ''
-    sed -i 's/,--version-script=.*$//' Makefile
-  '';
+  mesonFlags = [
+    "-Denable-wayland=false"
+    "-Dxkb-config-root=${xkeyboard_config}/etc/X11/xkb"
+    "-Dx-locale-root=${libX11.out}/share/X11/locale"
+  ];
 
   meta = with stdenv.lib; {
     description = "A library to handle keyboard descriptions";
-    homepage = http://xkbcommon.org;
+    homepage = https://xkbcommon.org;
     license = licenses.mit;
     maintainers = with maintainers; [ garbas ttuegel ];
     platforms = with platforms; unix;

